src/app/formControls/ObjectSchemaFormControl.ts

```diff
@@ -0,0 +1,51 @@
+import { FormArray, FormBuilder, FormGroup } from '@angular/forms';
+import { ObjectSchemaModel } from '../models/ObjectSchemaModel';
+
+/**
+ * Form control contains the mapping between the form and the backing model 
+ */
+export class ObjectSchemaFormControl {
```

Comment:
Should inherit from `AbstractCapabilityFormControl<T>`. Then you won't need your own instance of `subscribeModelToForm()`.

Reply:
Almost did this actually, was just thinking that it's not really a capability model though and T has to implement ICapabilityModel

Reply:
That's fair but really it could just be a naming thing. I recently had to remove the "name" property from the base model because name doesn't exist on Interface. This might ultimately lead to renaming of the base classes.